### PR TITLE
OGM-1006 Neo4j mass indexer embedded

### PR DESCRIPTION
**Pull request by DavideD**
https://hibernate.atlassian.net/browse/OGM-1006

### CODE REVIEW

**Comment by DavideD** [comment]
@Sanne or @gsmet  can you have a look at this?

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java [hunk 1]
```diff
@@ -467,6 +467,7 @@ private static String initFindEntitiesQuery(EntityKeyMetadata entityKeyMetadata,
 		queryBuilder.append( ENTITY );
 		queryBuilder.append( ":" );
 		appendLabel( entityKeyMetadata, queryBuilder );
+		queryBuilder.append( ")" );
```
Comment (gsmet) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java]: I'm surprised we only find this now. Don't we have a unit test exercising this?

**Comment by gsmet** [comment]
Typos in the commit messages:
s/retrieveal/retrieval/
s/depedendencies/dependencies/

**Comment by Sanne** [comment]
merged

File: neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java (hunk 1 above)
Comment (DavideD) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java]: Now that you mention it I think we missed the tests for the foreach method. It's been introduced for the mass indexer so it is only tested if we can run with the dialect.

I will create a JIRA for that

Comment (Sanne) [inline neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/dialect/impl/BaseNeo4jEntityQueries.java]: Indeed!

**Pull request closed without merging**